atlp-rwanda/e-commerce-bitcrafters-bn
# #187364726 -Fixing user registration validation and removing token while user signed up

**What does this PR do?**
This PR fixes the user registration process by making sure that user details are checked properly before signing up by using middlleware. It also removes the step where a special code (token) is created during signup.
 **Description of Task to be completed?**
We're adding a check to ensure that user details are correct before they sign up. Also, we're getting rid of the step where we create a special code when someone signs up.
 **How should this be manually tested?**

1. Go to the sign-up endpoint by using swagger  docs  or postman.
2. Try signing up with correct details.
3. Make sure you can sign up without any issues and without getting any special code.
4. Try signing up with incorrect details.
5. Make sure you get a message if you try signing up with wrong details.

 **Any background context you want to provide?**
Before this change, there wasn't a proper check to make sure user details were correct when signing up but using middleware. Also, a special token was created during signup, which isn't needed anymore.
**What are the relevant pivotal tracker/Trello stories?**
[#187364726]
**Screenshots (if appropriate)**
1) User tried to insert the invalid data.
![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/218c17e9-5e17-4def-a777-14797901936a)

2)Valid data and success message

![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/ffe29852-8879-44d9-89a0-b67530876873)
